### DISCUSSION TITLE
How create a piechart out of a column?

### DISCUSSION THREAD

**Discussion by lapwat** [grafana/grafana-csv-datasource#90] [Q&A] [answered]
I successfully imported my datasource, headers are recognized.

h1,h2
v1,v3
v1,v3
v2,v3

I would like to make a piechart out of one column. It should count all different values in this column.

For the data above, if I select column h1, it should return a piechart with v1:66% and v2:33%.

I am stuck at the point where it successfully count all values but does not make a difference between v1 and v2.

Can you help me achieve that ?

**Comment by marcusolsson** [comment]
Hi are you using the new or the old Pie chart panel? I believe the old Pie chart only supports time series data. In the new Pie chart, you should be able to select **All values**, to support the format you describe.

<img width="338" alt="Screenshot 2021-06-14 at 14 31 46@2x" src="https://user-images.githubusercontent.com/8396880/121892596-43ca2900-cd1d-11eb-8839-d1f438883d5c.png">

**Comment by lapwat** [comment]
Thanks for the quick response.

Selecting "All values" displays an empty Panel. 

![2021-06-14-143549_907x861_scrot](https://user-images.githubusercontent.com/7073651/121893134-e8e50180-cd1d-11eb-92f5-5c41f5846e59.png)

**Reply by marcusolsson** [reply]
The Pie chart needs a String field and a Number field when using All values, but looks like you're missing a Number field in your schema?

**Reply by lapwat** [reply]
I don't have any Number column, I would just like to count all rows with value "v1" in column "h1".

**Reply by lapwat** [reply]
In my case it could be, count all rows with value "Xiaomi" in column "manufacturer".

**Reply by marcusolsson** [reply]
Oh, I see! I think your best bet would be to a combination of Transformations to create the Number field first. 

Something like this maybe:

<img width="862" alt="Screenshot 2021-06-14 at 15 14 24@2x" src="https://user-images.githubusercontent.com/8396880/121898005-3a43bf80-cd23-11eb-8fb8-deff20c622d1.png">

**Comment by lapwat** [comment, marked as answer]
Thanks for enlightening me, I succeeded with the same technique.

I didn't need to add a new field.

![2021-06-14-155316_718x120_scrot](https://user-images.githubusercontent.com/7073651/121903556-b7bdfe80-cd28-11eb-9af6-eadc324b0f02.png)

**Reply by marcusolsson** [reply]
Thanks, I learned something new as well!
